barman/backup.py

```diff
@@ -579,6 +579,15 @@ def backup(self, wait=False, wait_timeout=None):
                 self.server, backup_id=datetime.datetime.now().strftime("%Y%m%dT%H%M%S")
             )
             backup_info.set_attribute("systemid", self.server.systemid)
+            if self.server.config.backup_compression is not None:
+                backup_info.set_attribute(
+                    "compression", self.server.config.backup_compression
+                )
+            elif self.server.config.backup_compression_level is not None:
```

Comment:
I don't think it should be the job of the BackupManager to know this detail which concerns the way pg_basebackup interprets its arguments.

It would be better to just set `backup_compression` in server.config at an earlier stage, ideally in an abstraction which sits somewhere on top of pg_basebackup.